### PR TITLE
MRG: Check limit monotonicity

### PR DESCRIPTION
**Pull request by larsoner**
A small tweak to ensure we check that `fmin < fmid < fmax`.

### CODE REVIEW

**Comment by larsoner** [comment]
CircleCI is going to take hours because it's queued behind `nipype` builds, but in the meantime AppVeyor and Travis are happy, so ready for review/merge from my end

**Comment by larsoner** [comment]
I'll merge in a couple days unless anyone wants to take a look

File: surfer/viz.py [hunk 1]
```diff
@@ -1064,19 +1064,21 @@ def add_data(self, array, min=None, max=None, thresh=None,
...
+                max = np.abs(center - array).max() if array.size > 0 else 1
         if mid is None:
             mid = (min + max) / 2.
+        _check_limits(min, mid, max, extra='')
 
         # Create smoothing matrix if necessary
         if len(array) < self.geo[hemi].x.shape[0]:
             if vertices is None:
-                raise ValueError("len(data) < nvtx: need vertices")
+                raise ValueError("len(data) < nvtx (%s < %s): need vertices"
```
Comment (mwaskom) [inline surfer/viz.py]: I know you haven't introduced it here, but I actually have no idea what "need vertices" means in this error message.

File: surfer/viz.py [hunk 2]
```diff
@@ -2751,6 +2753,16 @@ def _scale_sequential_lut(lut_table, fmin, fmid, fmax):
     return lut_table_new
 
 
+def _check_limits(fmin, fmid, fmax, extra='f'):
+    """Check for monotonicity."""
+    if fmin >= fmid:
+        raise ValueError('%smin must be < %smid, got %s >= %s'
```
Comment (mwaskom) [inline surfer/viz.py]: `%s` with floats is going to make this a message with a lot of decimal values, right? Maybe add a formatting code to improve signal to noise?

**Comment by mwaskom** [comment]
Just made a couple small comments. Any idea how these changes lose 4% coverage?

**Comment by larsoner** [comment]
> Any idea how these changes lose 4% coverage?

I assume it's just coveralls being buggy, there seem to be a lot of false alarms with Coveralls

**Comment by mwaskom** [comment]
> I assume it's just coveralls being buggy, there seem to be a lot of false alarms with Coveralls

Yeah ... I've found that `codecov` seems to work a lot better. Anyway, not a blocker here.

**Comment by larsoner** [comment]
@mwaskom updated to fix text and add `codecov` (couldn't resist)

**Pull request merged**